src/ProtocolStreamWriter.php

```diff
@@ -25,6 +25,16 @@ public function __construct($output)
      */
     public function write(Message $msg)
     {
-        fwrite($this->output, (string)$msg);
+        $data = (string)$msg;
+        $msgSize = strlen($data);
+        $totalBytesWritten = 0;
+
+        while ($totalBytesWritten < $msgSize) {
+            $bytesWritten = @fwrite($this->output, substr($data, $totalBytesWritten));
+            if ($bytesWritten === false) {
+                throw new Error('Could not write message.');
+            }
+            $totalBytesWritten += $bytesWritten;
+        }
```

Comment:
Could you explain a bit more what you are doing here? As I understand we block until everything is written out, is this intended? Should we use a setTimeout or nextTick here?

Also, `Error` is reserved for PHP internal errors, should be `Exception`